src/NumericalAlgorithms/Interpolation/IrregularInterpolant.cpp

```diff
@@ -22,25 +48,70 @@ template <>
 Matrix interpolation_matrix(
     const Mesh<1>& mesh,
     const tnsr::I<DataVector, 1, Frame::Logical>& points) noexcept {
-  // For 1d interpolation matrix, simply return the legendre-gauss-lobatto
-  // matrix.
+  if (mesh.basis()[0] == Spectral::Basis::FiniteDifference) {
+    const auto source_xi = logical_coordinates(mesh);
+    const auto number_of_source_points = mesh.number_of_grid_points();
+    std::vector<double> xi_source(number_of_source_points);
+    for (size_t s = 0; s < number_of_source_points; ++s) {
+      xi_source[s] = get<0>(source_xi)[s];
+    }
+    const auto number_of_target_points = get<0>(points).size();
+    Matrix result(number_of_target_points, mesh.number_of_grid_points());
+    for (size_t p = 0; p < number_of_target_points; ++p) {
+      const double xi_target = get<0>(points)[p];
+      const auto stencil = fd_stencil(xi_source, xi_target);
+      for (size_t i = 0; i < number_of_source_points; ++i) {
+        result(p, i) = stencil[i];
+      }
+    }
+    return result;
+  }
+
+  // Not FD, so use spectral interpolation
   return Spectral::interpolation_matrix(mesh, get<0>(points));
 }
 
 template <>
 Matrix interpolation_matrix(
     const Mesh<2>& mesh,
     const tnsr::I<DataVector, 2, Frame::Logical>& points) noexcept {
+  const auto number_of_target_points = get<0>(points).size();
+  Matrix result(number_of_target_points, mesh.number_of_grid_points());
+
+  if (mesh.basis()[0] == Spectral::Basis::FiniteDifference) {
+    const auto source_xi = logical_coordinates(mesh);
+    std::vector<double> xi_source(mesh.extents(0));
+    std::vector<double> eta_source(mesh.extents(1));
+    for (size_t i = 0; i < mesh.extents(0); ++i) {
+      xi_source[i] = get<0>(source_xi)[i];
+    }
+    for (size_t j = 0; j < mesh.extents(1); ++j) {
```

Comment:
if `mesh.extents(1) == mesh.extents(0)` you could save all the copying. Probably not a huge thing, but switch to DataVector and avoiding the copy would be nice. Up to you :)